django/contrib/admin/static/admin/js/selectfilter.js

```diff
@@ -0,0 +1,345 @@
+(function($) {
+    var settings = {
+        'name': '',
+        'verboseName': '',
+        'stacked': 0,
+        refresh_icons: function(selector) {
+            var available = selector.data('available');
+            var chosen = selector.data('chosen');
+            var is_from_selected = available.children().filter(':selected').length > 0;
+            var is_to_selected = chosen.children().filter(':selected').length > 0;
+            // Active if at least one item is selected
+            selector.find('.selector-add').toggleClass('active', is_from_selected);
+            selector.find('.selector-remove').toggleClass('active', is_to_selected);
+            // Active if the corresponding box isn't empty
+            selector.find('.selector-chooseall').toggleClass('active', available.find('option').length > 0);
+            selector.find('.selector-clearall').toggleClass('active', chosen.find('option').length > 0);
+        },
+        refresh_state: function(selector) {
+            settings.refresh_icons(selector);
+            var available = selector.data('available');
+            var chosen = selector.data('chosen');
+
+            // Re-generate the content of the actual, hidden, box.
+            var actualBox = selector.data('actualBox');
+            var actualOptions = actualBox.children();
+
+            // De-select all options.
+            available.children().each(function(){
+                var option = actualOptions[$(this).attr('data-index')];
+                option.selected = false;
+            });
+
+            // Select all the same options as those from the 'to' box.
+            chosen.children().each(function(){
+                var option = actualOptions[$(this).attr('data-index')];
+                option.selected = true;
+            });
+        },
+        move: function(from, to, all) {
```

Comment:
comments describing the purpose of the method and its parameters would be helpful